xtp/src/libxtp/dftengine/dftengine.cc

```diff
@@ -510,129 +521,404 @@ bool DFTEngine::EvaluateActiveRegion(Orbitals& orb) {
+      if (partOfActive == true) {
+        activemol_.push_back(orb.QMAtoms()[atom_num]);
+        // if active append the atom to the new molecule. Also increment active
+        // basis function size and start indices
+        start_indices_activemolecule.push_back(start_idx_activemolecule);
+        start_idx_activemolecule += numfuncpatom[atom_num];
+        numofactivebasisfunction += numfuncpatom[atom_num];
+      } else {
+        std::vector<const AOShell*> inactiveshell =
+            aobasis.getShellsofAtom(orb.QMAtoms()[atom_num].getId());
+        bool loop_break = false;
+        for (const AOShell* shell : inactiveshell) {
+          for (Index shell_fn_no = shell->getStartIndex();
+               shell_fn_no < shell->getStartIndex() + shell->getNumFunc();
+               shell_fn_no++) {
+            if (MnP[shell_fn_no] > truncation_threshold_) {
+              activeatoms.push_back(atom_num);
+              borderatoms.push_back(atom_num);  // push this index to border
+                                                // atoms
+              activemol_.push_back(orb.QMAtoms()[atom_num]);
+              loop_break = true;  // if any function in the whole atom satisfies
+                                  // we break loop to check next atom
+              start_indices_activemolecule.push_back(start_idx_activemolecule);
+              start_idx_activemolecule += numfuncpatom[atom_num];
+              numofactivebasisfunction += numfuncpatom[atom_num];
+              break;
+            }
+          }
+          if (loop_break) break;
+        }
+      }
+      start_idx += numfuncpatom[atom_num];
+    }
 
-    // get the XC contribution for the updated active density matrix
-    Mat_p_Energy xc_active = vxcpotential.IntegrateVXC(ActiveDensityMatrix);
-    double E_xc_active = xc_active.energy();
+    // Sort atoms before you cut the Hamiltonian
+    sort(activeatoms.begin(), activeatoms.end());
+    sort(borderatoms.begin(), borderatoms.end());
+    std::cout << std::endl
+              << "Active + Border Molecule Size = " << activeatoms.size()
+              << std::endl;
+    std::cout << std::endl
+              << "Border Molecule Size = " << borderatoms.size() << std::endl;
+
+    Eigen::MatrixXd H_embedding =
+        H0.matrix() + v_embedding + levelshift_ * ProjectionOperator;
+
+    if (borderatoms.size() != 0) {
+      Eigen::MatrixXd BorderMOs;
+      for (Index lmo_index = 0; lmo_index < InitialInactiveMOs.cols();
+           lmo_index++) {
+        double mullikenpop_lmo_borderatoms = 0;
+        for (Index borderatom : borderatoms) {
+          Index start = start_indices[borderatom];
+          Index size = numfuncpatom[borderatom];
+          mullikenpop_lmo_borderatoms +=
+              (InitialInactiveMOs.col(lmo_index) *
+               InitialInactiveMOs.col(lmo_index).transpose() * overlap.Matrix())
+                  .diagonal()
+                  .segment(start, size)
+                  .sum();
+        }
```

Comment:
Should the 0.25 be hardcoded?

Reply:
Reason written as a comment in code for the user to understand